### ISSUE TITLE
Ignore HugeParam or other settings for external interface

### ISSUE THREAD

**Issue by thespags** [go-critic/go-critic#1521]
If I'm binding my type to an external interface, e.g. slog.Handler, I don't own the ability to change
```Handle(ctx context.Context, record slog.Record) error```
to a pointer.

If possible, it would be nice if gocritic could detect that my custom handler implements slog.Handler and ignore. 
Or at least provide a list to ignore a set of interfaces.

**Comment by AlekSi** [comment]
It would be great if it recognized this idiom:

```go
var (
	_ slog.Handler = (*Handler)(nil)
)
```

**Comment by cristaloleg** [comment]
Sounds good, do you have a (few) code samples that should be ignored by the `hugeParam` chechker?

**Comment by thespags** [comment]
This is the example I had
```

type TraceHandler struct {
	next slog.Handler
}

var _ slog.Handler = (*TraceHandler)(nil)


func (h *TraceHandler) Enabled(ctx context.Context, level slog.Level) bool {
	return h.next.Enabled(ctx, level)
}

//nolint:gocritic // slog.Handler interface requires passing by value.
func (h *TraceHandler) Handle(ctx context.Context, record slog.Record) error {
      ... do stuff ...
	return h.next.Handle(ctx, record)
}


func (h *TraceHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
	return &TraceHandler{next: h.next.WithAttrs(attrs)}
}


func (h *TraceHandler) WithGroup(name string) slog.Handler {
	return &TraceHandler{next: h.next.WithGroup(name)}
}
```

**Comment by AlekSi** [comment]
My example is pretty much the same 😆
